v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/dialectadapter/mysql/MysqlDialectAdapter.java

```diff
@@ -151,7 +151,8 @@ public ImmutableList<String> discoverTables(
     } catch (SQLException e) {
       logger.error(
           String.format(
-              "Sql exception while discovering table list for datasource=%s", dataSource, e));
+              "Sql exception while discovering table list for datasource=%s cause=%s",
```

Comment:
nit: you don't need to use String.format with slf4j. You can just use `{}` for string placeholders, like

```
 logger.error("Sql exception while discovering table list for datasource={}", dataSource, e)
```

We should also log the exception not in the String, but separately as above to get the stack trace.

Reply:
yup, this entire file needs a cleanup for slf4 formatted logging.  Most of the files in the reader follow the sl4j convention. I will clean up this file in a followup.